### PR TITLE
feat(actions) Add additional encrypt/decrypt actions. Allow session create to use different users.

### PR DESCRIPTION
**Pull request by cball**
This PR is tackling two very important issues.
1) A complex config for the consuming app's Fastlane setup
2) A way to authorize a session and then later login with different accounts.

## Changes
- Simplify `create_fastlane_session`. It now re-uses some ENV vars, and prompts for the user to release as.
- Add `add_app_var` action to add a single env var
- Add `decrypt_android_keystore` and `generate_android_keystore` to make working with keystores easier
- Add `encrypt_app_vars` and `decrypt_app_vars` to handle writing / restoring ENV. `decrypt_app_vars` takes an optional namespace which allows us to override for `alpha`, `beta`, or `release`
- Add `encrypt_google_play_credentials` and `decrypt_google_play_credentials` to handle writing/restoring the Google Play JSON

### CODE REVIEW

**Comment by code-jenn-or** [comment]
Can we please update the readme to include which `.env` files should have which variables?
i.e. `ROOT/.env` : app specific variables like API url, etc
`ROOT/ios/fastlane/.env` : list of vars that should be in this one
`ROOT/android/fastlane/.env` ....
`ROOT/fastlane/.env`

File: lib/fastlane/plugin/react_native_release/actions/add_app_var.rb [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+      def self.details
+        # Optional:
+        # this is your chance to provide a more detailed description of this action
+      end
+
+      def self.available_options
+        [
+          FastlaneCore::ConfigItem.new(key: :namespace,
+                                       env_name: "FL_ADD_APP_VAR_NAMESPACE",
+                                       description: "What namespace should we use? (beta, release, ENTER = root)", # a short description of this parameter
```
Comment (ryanpatk) [inline lib/fastlane/plugin/react_native_release/actions/add_app_var.rb]: do we want `alpha` ?

File: lib/fastlane/plugin/react_native_release/actions/add_app_var.rb [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+        # If your method provides a return value, you can describe here what it does
+      end
+
+      def self.details
+        # "Saves the current vars in android/fastlane/.env and ios/fastlane/.env"
+      end
+
+      def self.authors
+        # So no one will ever forget your contribution to fastlane :) You are awesome btw!
+        ["cball", "isaiahgrey93"]
```
Comment (rynatk) [inline lib/fastlane/plugin/react_native_release/actions/add_app_var.rb]: 🔥

File: lib/fastlane/plugin/react_native_release/actions/decrypt_app_vars.rb [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+      def self.details
+        # Optional:
+        # this is your chance to provide a more detailed description of this action
+      end
+
+      def self.available_options
+        [
+          FastlaneCore::ConfigItem.new(key: :namespace,
+                                       env_name: "FL_DECRYPT_APP_VARS_NAMESPACE", # The name of the environment variable
+                                       description: "What namespace should we use? (beta, release, ENTER = root)", # a short description of this parameter
```
Comment (ryanpatk) [inline lib/fastlane/plugin/react_native_release/actions/decrypt_app_vars.rb]: same comment about `alpha`

File: lib/fastlane/plugin/react_native_release/actions/react_native_release.rb [hunk 4]
```diff
@@ -4,45 +4,39 @@
...
 
         if is_beta
-          tag_prefix = 'betas'
-          base_branch = params[:alpha_branch]
-          target_branch = params[:beta_branch]
-        else
-          tag_prefix = 'releases'
-          base_branch = params[:beta_branch]
-          target_branch = params[:production_branch]
+          # TODO: ohter branches
```
Comment (ryanpatk) [inline lib/fastlane/plugin/react_native_release/actions/react_native_release.rb:25]: `ohter`

File: lib/fastlane/plugin/react_native_release/actions/react_native_release.rb [hunk 5]
```diff
@@ -4,45 +4,39 @@
...
-        other_action.git_pull
+        # TODO: flows
 
         # Cut a fresh branch unless this is a hotfix
-        if !is_hotfix
-          # delete an existing branch if we have one
-          sh "git show-ref #{target_branch}" do |status|
-            sh "git branch -D #{target_branch}" if status.success?
-          end
+        # if !is_hotfix
```
Comment (ryanpatk) [inline lib/fastlane/plugin/react_native_release/actions/react_native_release.rb:32]: # TODO: hotfix

**Review by ryanpatk — APPROVED** [review]
looks good to me. Maybe some additional comments in the 
`TODO: flows` section to help jog anyone's memory when they come back there

nice work!

File: lib/fastlane/plugin/react_native_release/actions/react_native_release.rb (hunk 5 above)
Comment (ryanpatk) [reply lib/fastlane/plugin/react_native_release/actions/react_native_release.rb:32]: lol didnt mean to do markdown here

Comment (cball) [reply lib/fastlane/plugin/react_native_release/actions/react_native_release.rb:32]: added as #24

File: lib/fastlane/plugin/react_native_release/actions/react_native_release.rb (hunk 4 above)
Comment (cball) [reply lib/fastlane/plugin/react_native_release/actions/react_native_release.rb:25]: will address in #24

File: lib/fastlane/plugin/react_native_release/actions/add_app_var.rb [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
```
Comment (mcavaliere) [inline lib/fastlane/plugin/react_native_release/actions/add_app_var.rb:82]: 👍

File: lib/fastlane/plugin/react_native_release/actions/react_native_release.rb [hunk 7]
```diff
@@ -4,45 +4,39 @@
...
       def self.run(params)
         require 'fastlane/plugin/android_versioning'
-
         other_action.create_fastlane_session
 
-        target = UI.select "Select a release type:", VALID_TARGETS
-        is_beta = target.include?('beta')
-        is_hotfix = params[:hotfix] === true
+        release_type = params[:release_type] || UI.select("Select a release type:", VALID_RELEASE_TYPES)
+        is_beta = release_type.include?('beta')
```
Comment (mcavaliere) [inline lib/fastlane/plugin/react_native_release/actions/react_native_release.rb:16]: Not important at all, but these three lines (and the following conditional) could be done metaprogram-y  to set the state for more release types in the future.

**Review by mcavaliere — APPROVED** [review]
My Fastlane skill is wayyyy out of date, but FWIW looks amazing. Two small unimportant comments.

File: lib/fastlane/plugin/react_native_release/actions/react_native_release.rb (hunk 7 above)
Comment (cball) [reply lib/fastlane/plugin/react_native_release/actions/react_native_release.rb:16]: totally agree. I want to rework this whole action in a followup PR to properly support master-based and gitflow-based workflows (and hotfixes). I'll try and clean up there as well.

File: lib/fastlane/plugin/react_native_release/actions/add_app_var.rb (hunk 6 above)
Comment (cball) [reply lib/fastlane/plugin/react_native_release/actions/add_app_var.rb:82]: look forward to increasing this array size 😄

**Pull request merged**